### DISCUSSION TITLE
Conversations and FSM

### DISCUSSION THREAD

**Discussion by hurek** [Rapptz/discord.py#5964] [Q&A]
Hello! Please tell me if there is support for FSM and States in discord, so that you can simulate dialogues and something like that. (for example, this is useful for registering users and subscribing to some notifications).

**Comment by Rapptz** [comment]
I'm not sure this has much to do with the library itself.

You can employ a various numbers of FSM techniques using Python itself. Alternatively, async functions are basically another form of FSM. 

The scope of this topic is a bit too wide to be answered properly.

**Reply by hurek** [reply]
Sorry, I'm novice in python. Can you advice me easiest way to implement FSM with discordpy? Thank you!

**Reply by Rapptz** [reply]
It's way outside the scope of the support you'll get here. You're better off using google. This has little to do with the library and more to do with Python so start there.
